src/pricecypher/rest.py

```diff
@@ -0,0 +1,257 @@
+import json
+import requests
+from time import sleep
+from random import randint
+from datetime import datetime
+from marshmallow import Schema
+
+from .exceptions import PriceCypherError, RateLimitError
+
+UNKNOWN_ERROR = 'pricecypher.sdk.internal.unknown'
+
+
+class RestClientOptions(object):
+    """
+    Configuration object for RestClient.
+    Used for configuring additional RestClient options, such as rate-limit retries.
+
+    :param float or tuple[float,float] timeout: (optional) Change the requests connect and read timeout (in seconds).
+        Pass a tuple to specify both values separately or a float to set both to it.
+        (defaults to 300.0 (5 minutes) for both)
+    :param int retries: (optional) In the event an API request returns a 429 response header (indicating rate-limit
+        has been hit), the RestClient will retry the request this many times using an exponential backoff strategy,
+        before raising a RateLimitError exception.
+        (defaults to 3)
+    """
+
+    def __init__(self, timeout=None, retries=None):
+        self.timeout = 300.0
+        self.retries = 3
+
+        if timeout is not None:
+            self.timeout = timeout
+
+        if retries is not None:
+            self.retries = retries
```

Comment:
```suggestion
    def __init__(self, timeout=300.0, retries=3):
        self.timeout = timeout
        self.retries = retries
```

Reply:
This does change the functionality somewhat. With the current version, we can guarantee the values to not be None, whereas with the suggested version this is not guaranteed.